compiler/cpp/src/reader/Reader.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Marc on 23.05.2021.
+//
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <iostream>
+#include "../exception/UnexpectedCharException.h"
+
+class Reader {
+private:
+    // Constant inputs
+    std::string fileInput;
+    unsigned int maxLookahead;
+
+    // Working variables
+    int curChar = 0;
+    int inputStringPos = 0;
+    unsigned int lineNum = 1;
+    unsigned int colNum = 0;
+public:
+    explicit Reader(): maxLookahead(0) {};
+    Reader(std::string fileInput, unsigned int maxLookahead);
+
+    void advance();
+    void expect(int);
+    void expectMultiple(const std::string&);
+    int getLookahead() const;
+    std::string getMaxLookahead();
+    unsigned int getLineNum() const;
+    unsigned int getColNum() const;
```

Comment:
Did you also implement `toPosStringWithCaret()` somewhere?

Reply:
Nope. And there are currently no plans to to so. I think the performance loss would be more significant then the gain of convinience of the user.

Reply:
ok